### ISSUE TITLE
Comments features not showing properly

### ISSUE THREAD

**Issue by Dr-Left** [academicpages/academicpages.github.io#2655]
**Describe the bug**
When I want to use comments providers, such as `giscus`, the page is not behaving as what I think. I can only see a “LEAVE A COMMENT” label at the bottom of the blog posts but no actual widgets.

**To Reproduce**
Follow the [tutorial by Lazy Ren](https://lazyren.github.io/devlog/use-utterances-for-jekyll-comments.html#my-commentshtml)

**Debugging**
I believe the problem is that the `_layouts/default.html` doesn't include the `/comments-providers/scripts.html`.
I added `{% include comments-providers/scripts.html %}` to `_includes/scripts.html` and the problem is solved.
Is this a bug? If so I hope you could add the line to the file to solve this problem for others.

Before:
<img width="571" alt="image" src="https://github.com/user-attachments/assets/0c9567e3-d471-4c19-b2eb-bfa5773c5cd0">

After:
<img width="600" alt="image" src="https://github.com/user-attachments/assets/9f019d3a-dc82-42d6-ad17-e5acdf1d2132">

**Comment by rjzupkoii** [comment]
This does look like a bug, but there are other comment providers, so we would need to check to make sure adding `/comments-providers/scripts.html` to `_layouts/default.html` doesn't break things. Off-hand I suspect it would because the comments should only be rendering in places where they make sense such as the blog posts, as opposed to throughout the site, which edits to some of the default pages can tdo.

**Comment by Dr-Left** [comment]
Yeah, I believe the code in `/comments-providers/scripts.html` take care of the problem you mentioned: only add comments section to blog posts, etc. instead of throughout the site. Check this line in `/comments-providers/scripts.html`:

```html
{% if site.comments.provider and page.comments %}
```

If the `_config.yaml` does assign a true value to the `comments` field of a certain page, i.e. `page.comments == true`, then the comments section would appear; if not, it would not appear.

**Comment by huanfachen** [comment]
Thanks for this post. I tested adding this line in scripts.html on my personal website and this method worked.
